### ISSUE TITLE
Accuracy of ERF predictions for unseen observations

### ISSUE THREAD

**Issue by William-M-Ricardo** [nicolagnecco/erf#8]
Hello,

Thank you for developing the ERF package; it has been very useful in my work! I have a question regarding the accuracy of predictions when the model encounters new, unseen observations.

Given that ERF is based on random forests, I understand that it is generally good at interpolation but may struggle with extrapolation. My concern is that if a new observation vector (in the test set) is far from or not present in the training set, would the predicted extreme quantiles be reliable?

Could you provide any guidance on how ERF handles such cases, and whether there are any techniques or strategies you would recommend to improve prediction accuracy for unseen observations, especially in the context of extreme quantile regression?

Thank you for your time, and I look forward to your insights!

Best regards,  
William

**Comment by nicolagnecco** [comment]
Hi William,

Thank you for your question. In general, extrapolation with non-parametric models is a delicate topic. In particular, without any assumptions on how the model extrapolates beyond the observed support, it is hard to give any guarantee.
For example, if you knew (from, e.g., domain knowledge) that the function is likely to extrapolate flatly, then classical random forests, and therefore ERF, would work well. On the other hand, if you suspect that linear extrapolation is more reasonable, one might want to consider local linear random forests (e.g., https://arxiv.org/pdf/1807.11408).

As a last note, ERF is geared towards extrapolation in the response. For a fixed x in the observed support, ERF estimates quantiles beyond the range of observed data (in other words, ERF predictions are conditional on x). This extrapolation is different from the one in the predictor space you mention.
